x/superfluid/keeper/migrate.go

```diff
@@ -97,21 +99,13 @@ func (k Keeper) migrateSuperfluidBondedBalancerToConcentrated(ctx sdk.Context,
 	// If all shares are being migrated, this deletes the connection between the gamm lock and the intermediate account, deletes the synthetic lock, and burns the synthetic osmo.
 	intermediateAccount := types.SuperfluidIntermediaryAccount{}
 	var gammLockToMigrate *lockuptypes.PeriodLock
-	if isPartialMigration {
-		// Note that lock's id is different from the originalLockId since it was split.
-		// The original lock id stays in gamm.
-		intermediateAccount, gammLockToMigrate, err = k.partialSuperfluidUndelegateToConcentratedPosition(ctx, sender.String(), originalLockId, sharesToMigrate)
-		if err != nil {
-			return 0, sdk.Int{}, sdk.Int{}, sdk.Dec{}, 0, 0, 0, err
-		}
-	} else {
-		// Note that lock's id is the same as the originalLockId since all shares are being migrated
-		// and old lock is deleted
-		gammLockToMigrate = preMigrationLock
-		intermediateAccount, err = k.SuperfluidUndelegateToConcentratedPosition(ctx, sender.String(), originalLockId)
-		if err != nil {
-			return 0, sdk.Int{}, sdk.Int{}, sdk.Dec{}, 0, 0, 0, err
-		}
+
+	// Note that lock's id is the same as the originalLockId since all shares are being migrated
+	// and old lock is deleted
+	gammLockToMigrate = preMigrationLock
```

Comment:
Why do we reassign here when there is no more split option?

Reply:
yup not necessary